e2e/test/iothub/service/PurgeMessageQueueE2ETests.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Azure.Devices.E2ETests.iothub.service
+{
+    /// <summary>
+    /// E2E test class for PurgeMesageQueue.
+    /// </summary>
+    [TestClass]
+    [TestCategory("E2E")]
+    [TestCategory("IoTHub")]
+    public class PurgeMesageQueueE2ETests : E2EMsTestBase
+    {
+        [TestMethod]
+        public async Task PurgeMessageQueueOperation()
+        {
+            using Message testMessage = ComposeD2CTestMessage();
+            using var sc = new IotHubServiceClient(TestConfiguration.IoTHub.ConnectionString);
+            var deviceId = TestConfiguration.IoTHub.X509ChainDeviceName;
+            var expectedResult = new PurgeMessageQueueResult()
+            {
+                DeviceId = deviceId,
+                TotalMessagesPurged = 50
+            };
+            for (int i = 0; i < 50; ++i)
+            {
+                await sc.Messaging.SendAsync(deviceId, testMessage);
+            }
+            PurgeMessageQueueResult result = await sc.Messaging.PurgeMessageQueueAsync(deviceId, CancellationToken.None).ConfigureAwait(false);
+            Assert.AreEqual(expectedResult.DeviceId, result.DeviceId);
+            Assert.AreEqual(expectedResult.TotalMessagesPurged, result.TotalMessagesPurged);
+        }
+
+        private Message ComposeD2CTestMessage()
```

Comment:
You don't need to go so far as to create a message with all these detailed properties for this test to work. 

```csharp
return var new Message(Encoding.UTF8.GetBytes("some payload"));
```